### PR TITLE
add step0 for s2-L1C

### PR DESCRIPTION
**Pull request by gilliM**
add step0 for s2-L1C

Need changes of configuration similar to gmi_config.py:

- definition of an empty_asset_list:
EMPTY_ASSET_LIST = os.path.join("tools", "step0_empty_assets.csv")

- Addition of a step0_collection entry in the product dictionary.

- addition of a step0 dictionary.

Note that the example provided of S2 L1C require the definition of the new L1C product. So the whole dictionary can be copied from gmi_config.py

For the PRODUCT_NDVI_MAX_TOA, the temporal_coverage entry should be increase to 30 days. I kept at at 1 for development purposes.

### CODE REVIEW

**Comment by gilliM** [comment]
Screenshot from 2023-12-04 16-07-31
Screenshot from 2023-12-04 16-08-35

As a not here, this code processed 10 october 2023 without throwing error, and did create a quite large asset (~24GB for a single swath, highest ever observed on my side.)

**Comment by davidoesch** [comment]
Interesting 2023-10-01 : to calculate and save asset it takes 350h in your python code, to calculate and export TIFF via GEE JS we have 576 hours ....  see https://docs.google.com/spreadsheets/d/1V5cgHQicFL8ZCX7Ge7n-aB3DUtSCYZGTVE0NX9AwMpw/edit#gid=1260727456&range=F10 ( and there is even one quadarant missing it will be rather 700 hours) .

File: step0_processor.py [hunk 1]
```diff
@@ -0,0 +1,513 @@
+import ee
+from datetime import datetime, timedelta
+
+ee.Initialize()
+
+
+def generate_asset_mosaic_for_single_date(day_to_process: datetime, export_to_drive: bool = False,
```
Comment (davidoesch) [inline step0_processor.py]: Question: 
- This function is made for S2-L2A, and not for S2-L1C, so where would the switch be? ow r will it be a different function?
- Where are the switches as defined inhttps://code.earthengine.google.com/a08635b7f3c83b082c35023a72743ec4 line 46 to line 69. Does it makes sense to have them as well in config file in the product definition?

**Comment by gilliM** [comment]
It looks like I mixed up both product scripts... I had a first version for L2A and then the L1C javascript was produced. I adapted the script for L1C, but I missed the initial definitions. I will do a full review of it.

**Comment by gilliM** [comment]
Actually the line  https://github.com/swisstopo/topo-satromo/blob/636c50785bcf7793b7ec2c61c90e559eeb48e1b2/step0_processor.py#L150 comes from the original pipeline to write the per-original-tile json files. I indeed forgot to adapt the content, although it does not come from the javascript.

The comment https://github.com/swisstopo/topo-satromo/blob/636c50785bcf7793b7ec2c61c90e559eeb48e1b2/step0_processor.py#L251 will be corrected, although it was present un the TOA javascript file.

Which points in the script made you say that it's a L2A script and not a L1C ?

**Comment by gilliM** [comment]
It looks like you reviewed an intermediate commits within the PR. You should review the whole final PR version with all the commits.

**Comment by davidoesch** [comment]
> Actually the line
> 
> https://github.com/swisstopo/topo-satromo/blob/636c50785bcf7793b7ec2c61c90e559eeb48e1b2/step0_processor.py#L150
> 
> comes from the original pipeline to write the per-original-tile json files. I indeed forgot to adapt the content, although it does not come from the javascript.
> The comment
> 
> https://github.com/swisstopo/topo-satromo/blob/636c50785bcf7793b7ec2c61c90e559eeb48e1b2/step0_processor.py#L251
> 
> will be corrected, although it was present un the TOA javascript file.
> Which points in the script made you say that it's a L2A script and not a L1C ?

hmm seems I looked at on old version

**Comment by davidoesch** [comment]
I openend you PR in 
image
 fetched it

and now I have these files

image

Am I looking a the wrong  files?
image

**Comment by davidoesch** [comment]
> maybe there is some confusion. This function is for L1C. I've added L1C functions in satromo_processor.py https://github.com/swisstopo/topo-satromo/pull/18/files#diff-56d4bafe9ad687fc58aa82c9deda5c87fbbb7be629735f1a3c405cb61ccebd6bR681
> 
> I have not adapted process_NDVI_MAX and process_S2_LEVEL_2A in satromo_processor.py to handle the step0 pipeline. The configuration that i provide in gmi_config.py will only works for L1C products.

In this case I will try to set it up and keep it running with L1C- for the time beeing. The support for L2A will be in a next PR I assume.

**Pull request merged**